### ISSUE TITLE
mochaRunner: Cannot read property 'testLogPosition' of undefined'

### ISSUE THREAD

**Issue by BurtHarris** [alm-tools/alm#272]
I'm just starting with alm.    Trying to get integration with some mocha-based tests cases.   I've configured `alm.json` to run just one test file with low dependencies, but alm isn't recognizing the tests...

alm.json:

```
{
  "tests": {
    "include": [
      "./test/TestBitSet.ts"
    ]
  }
}
```

The console window output that seems relevant is:

```
DASHBOARD: (Please open in chrome) http://localhost:4444
TSCONFIG: All Good! C:/code/antlr4ts/tsconfig.json
[TSC] Started Initial Error Analysis: C:/code/antlr4ts/tsconfig.json
[TESTED] File count: 1
[TESTED] Testing
[TESTED] File count: 1
[LINT] No tslint configuration found.
[TSC] Started Initial Error Analysis: C:/code/antlr4ts/tsconfig.json
C:\Users\Burt_\AppData\Roaming\npm\node_modules\alm\src\server\workers\tested\runners\mochaRunner.js:180
        currentSuite.testLogPosition = suitePositions.find(function (s) { return s.title == currentSuite.description; }).testLogPosition;
                                                                                                                        ^

TypeError: Cannot read property 'testLogPosition' of undefined
    at getOrCreateSuite (C:\Users\Burt_\AppData\Roaming\npm\node_modules\alm\src\server\workers\tested\runners\mochaRunner.js:180:121)
    at C:\Users\Burt_\AppData\Roaming\npm\node_modules\alm\src\server\workers\tested\runners\mochaRunner.js:186:21
    at Array.forEach (native)
    at parseMochaJSON (C:\Users\Burt_\AppData\Roaming\npm\node_modules\alm\src\server\workers\tested\runners\mochaRunner.js:184:11)
    at ChildProcess.<anonymous> (C:\Users\Burt_\AppData\Roaming\npm\node_modules\alm\src\server\workers\tested\runners\mochaRunner.js:90:21)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
testedWorker worker restarting. Don't know why it stopped with code: 1
[TSC] Error Analysis Duration: 6s
[TSC] FileCount: 179  Errors: 0
```

**Comment by BurtHarris** [comment]
Seems like it would be nice if alm logged the command-line of the process spawned so I can easily debug it myself.

**Comment by BurtHarris** [comment]
It looks like this line of code in MochaRunner.ts is not robust to `Array.find()` failing to find a match:

https://github.com/alm-tools/alm/blob/master/src/server/workers/tested/runners/mochaRunner.ts#L204

**Comment by basarat** [comment]
Note to self: 
* clone https://github.com/tunnelvisionlabs/antlr4ts
* npm install
* add specified alm.json 

It works if `describe` nesting is removed. Investigating :rose:

**Comment by basarat** [comment]
> It looks like this line of code in MochaRunner.ts is not robust to Array.find() failing to find a match

Not designed to be robust as *it should work* and the fact that it errors is a good symptom to fix instead of siliently ignoring ;)

**Comment by basarat** [comment]
Doesn't Work: 
```
describe("Level.1 Suite", () => {
    describe("Level.2 Suite", () => {
        it("Level.2 test", ()=> {
        });
    })
    // it("Level.1 test", ()=> {
    // });
})
```

Works

```ts
describe("Level.1 Suite", () => {
    describe("Level.2 Suite", () => {
        it("Level.2 test", ()=> {
        });
    })
    it("Level.1 test", ()=> {
    });
})

```

PS: I get what you meant but it not being robust ... aka it has a bug. Investigating :rose:

**Comment by basarat** [comment]
Summary. We get the suite description from `const suiteDescription = test.fullTitle.substr(0, test.fullTitle.length - test.title.length).trim();`. If there is no test in an outer level the first suite description we see is `suiteouter suiteinner` and hence cannot match this description with any suite positions (there is no `suiteouter suiteinner` position, only `suiteouter` and `suiteinner` positions)

**Comment by basarat** [comment]
Fixed with `2.12.0` :rose: Thanks for you patience :heart:
